MapboxNavigation/NavigationView.swift

```diff
@@ -38,26 +38,133 @@ import MapboxDirections
-    weak var statusView: StatusView!
-    weak var resumeButton: ResumeButton!
-    // Vertically laid-out stack view below the instructions banner consisting of StatusView, NextBannerView, and LanesView.
-    weak var informationStackView: UIStackView!
-    // Vertically laid-out stack view below the information stack view ontop of the map view, docked
-    // to the top right, consisting of Overview, Mute, and Report button.
-    weak var floatingStackView: UIStackView!
-    weak var overviewButton: FloatingButton!
-    weak var muteButton: FloatingButton!
-    weak var reportButton: FloatingButton!
-    weak var rerouteReportButton: ReportButton!
-    weak var separatorView: SeparatorView!
+    lazy var bannerShowConstraints: [NSLayoutConstraint] = [
+        self.instructionsBannerView.topAnchor.constraint(equalTo: self.safeTopAnchor),
+        self.instructionsBannerContentView.topAnchor.constraint(equalTo: self.topAnchor)]
+    
+    lazy var bannerHideConstraints: [NSLayoutConstraint] = [
+        self.instructionsBannerView.bottomAnchor.constraint(equalTo: self.topAnchor),
+        self.instructionsBannerContentView.topAnchor.constraint(equalTo: self.instructionsBannerView.topAnchor)
+    ]
+    
+    lazy var endOfRouteShowConstraint: NSLayoutConstraint? = self.endOfRouteView?.bottomAnchor.constraint(equalTo: self.safeBottomAnchor)
+    
+    lazy var endOfRouteHideConstraint: NSLayoutConstraint? = self.endOfRouteView?.topAnchor.constraint(equalTo: self.bottomAnchor)
+    
+    lazy var endOfRouteHeightConstraint: NSLayoutConstraint? = self.endOfRouteView?.heightAnchor.constraint(equalToConstant: 260)
+    
+    private struct Images {
+        static let overview = UIImage(named: "overview", in: .mapboxNavigation, compatibleWith: nil)!.withRenderingMode(.alwaysTemplate)
+        static let volumeUp = UIImage(named: "volume_up", in: .mapboxNavigation, compatibleWith: nil)!.withRenderingMode(.alwaysTemplate)
+        static let volumeOff =  UIImage(named: "volume_off", in: .mapboxNavigation, compatibleWith: nil)!.withRenderingMode(.alwaysTemplate)
+        static let feedback = UIImage(named: "feedback", in: .mapboxNavigation, compatibleWith: nil)!.withRenderingMode(.alwaysTemplate)
+    }
+    
+    private struct Actions {
+        static let cancelButton: Selector = #selector(NavigationView.cancelButtonTapped(_:))
+    }
+    
+    private static let rerouteReportTitle = NSLocalizedString("REROUTE_REPORT_TITLE", bundle: .mapboxNavigation, value: "Report Problem", comment: "Title on button that appears when a reroute occurs")
+    
+    static let buttonSize = CGSize(width: 50, height: 50)
+    
+    lazy var mapView: NavigationMapView = {
+        let map: NavigationMapView = .forAutoLayout()
+        map.delegate = delegate
+        map.navigationMapDelegate = delegate
+        map.courseTrackingDelegate = delegate
+        map.styleURL = URL(string: "mapbox://styles/mapbox/navigation-guidance-day-v2")
```

Comment:
Does setting a custom styleURL on a `Style` class still work?

Reply:
This line was added because of it's equivalent in [the storyboard](https://github.com/mapbox/mapbox-navigation-ios/blob/master/MapboxNavigation/Resources/Base.lproj/Navigation.storyboard#L43), I'm not quite sure if it's even needed.